File: php/lib/steam/packets/A2A_SERVERQUERY_GETCHALLENGE_ResponsePacket.php
<?php
/**
 * @author Sebastian Staudt
 * @license http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package Steam Condenser (PHP)
 * @subpackage SteamPacket
 * @version $Id$
 */

require_once "steam/packets/SteamPacket.php";

/**
 * @package Steam Condenser (PHP)
 * @subpackage SteamPacket
 */
class A2A_SERVERQUERY_GETCHALLENGE_ResponsePacket extends SteamPacket
{	
	/**
	 * @param long $challengeNumber
	 */
	public function __construct($challengeNumber)
	{
		parent::__construct(SteamPacket::A2A_SERVERQUERY_GETCHALLENGE_RESPONSE_HEADER, $challengeNumber);
	}
	
	/**
	 * @return int
	 */
	public function getChallengeNumber()
	{
		return $this->contentData->rewind()->getLong(); 
	}
}
?>

File: ruby/steam/packets/a2a_serverquery_getchallenge_response_packet.rb
class A2A_SERVERQUERY_GETCHALLENGE_ResponsePacket < SteamPacket
  def initialize(challenge_number)
    super A2A_SERVERQUERY_GETCHALLENGE_RESPONSE_HEADER, challenge_number.unpack("V")[0]
  end
  
  def get_challenge_number
    return @content_data
  end
end
